### PR TITLE
Use tox's generative envlist

### PR DESCRIPTION
**Pull request by redapple**
Test all combinations of Python 2/3 and scrapy 1.0, 1.1, 1.2 and 1.3 branches.

I think we can probably get rid of this extra `-r{toxinidir}/requirements-tox.txt`, and dependencies from setup.py would be used (I added it because of duplicate dependencies on scrapy that tox was complaining about).
I'm far from a tox expert so suggestions are welcome.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-    -r{toxinidir}/requirements.txt
-    -r{toxinidir}/requirements-dev.txt
-commands = py.test {posargs}
-
-[testenv:py27-scrapy1.0.x]
-deps =
-    Scrapy<1.1
-    -r{toxinidir}/requirements-dev.txt
+[tox]
+envlist = py27, py27-scrapy{1.0,1.1,1.2,1.3}, py35, py36, {py35,py36}-scrapy{1.1,1.2,1.3}
```
Comment (chekunkov) [inline tox.ini]: why keep py27, py35 and py36 then?

File: requirements-tox.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+service-identity>=1.0.0
+demjson
+six
```
Comment (chekunkov) [inline requirements-tox.txt]: what's the purpose of this file if we use setup.py for deps?

File: tox.ini (hunk 1 above)
Comment (redapple) [inline tox.ini]: It's for up-to-date versions (when there's Scrapy>1.3), but today it's indeed equivalent to using scrapy1.3

Comment (redapple) [inline tox.ini]: (we'll soon be releasing scrapy 1.4)

File: requirements-tox.txt (hunk 2 above)
Comment (redapple) [inline requirements-tox.txt]: Done.

File: tox.ini (hunk 1 above)
Comment (chekunkov) [inline tox.ini]: anyway your change means that we will need to update tests with every minor Scrapy release, why keep duplicate env? if you want to keep `py*` envs let's maybe remove `py*-scrapy1.3` envs and add them once you release 1.4? `py*` envs will test latest stable version in this case and `py*-scrapy*` will test older minor releases.

Comment (redapple) [inline tox.ini]: Makes sense. Will update

Comment (redapple) [inline tox.ini]: Done.
